ReactQt/runtime/src/reactuimanager.cpp

```diff
@@ -128,8 +128,8 @@ void ReactUIManager::setChildren
 {
   //TODO: This is a simple implementation which fixes a broken example. It's not properly tested and may need revisiting
   QList<int> indices;
-  foreach(int i, childrenTags) {
-    indices.append(0); //0 is a default z-order in qml
+  for(int i = 0; i < childrenTags.size(); ++i) {
+    indices.append(i); //Incrementing per each child. Similar to iOS. See the comment in manageChildren
```

Comment:
@MaxRis, right, will be good to have support of Z index of  react components, but that will not solve this issue, since we actually don't have Z index set on js part.  
current issue is not caused by Z index directly - it's caused by using same 0 values as indices to set children into layout (in my understanding and as testing showed, caused by wrong order in the list of items - PR initial description). 
I'll adjust the solution to not use Z index. 

@vkjr, this code in PR works ok for redbox case. But anyway needs to be changed, and probably the final code will have 0 Z indices.